src/model/mod.rs

```diff
@@ -48,8 +50,145 @@ pub enum PlayingItem {
     Episode(show::FullEpisode),
 }
 
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct Id<'id> {
+    _type: Type,
+    id: &'id str,
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Display, Error)]
+pub enum IdError {
+    InvalidPrefix,
+    InvalidFormat,
+    InvalidType,
+    InvalidId,
+}
+
+impl std::fmt::Display for Id<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        f.write_str("spotify:")?;
```

Comment:
You could just use `write_fmt` here and undo deriving `AsRefStr` for `Type`.

Reply:
If I don't use `AsRefStr`, I'd have to use either `Display` or `ToString`. I have a hard "no" for `ToString`, as it allocates, and the whole point of this PR is to avoid unnecessary allocations. So if I remove `AsRefStr` I will end up implementing `Display` on `Type` anyway. Do you think it's better?

Reply:
By the way, since I already use `Display` instead of `ToString` for `Type`, maybe we should go to the end and use it instead of `ToString` for all other enums?